test/models/compute_resource_test.rb

```diff
@@ -126,8 +126,7 @@ def setup
 
   test '.providers returns merge of loaded builtin and registered providers' do
     ComputeResource.expects(:registered_providers).returns({'Libvirt' => 'Best::Provider::Libvirt', 'MyBest' => 'Best::Provider::MyBest'})
-    ComputeResource.expects(:supported_providers).returns({'Libvirt' => 'Foreman::Model::Libvirt', 'EC2' => 'Foreman::Model::EC2', 'GCE' => 'Foreman::Model::GCE'})
-    Fog::Compute.expects(:providers).twice.returns([:aws, :libvirt])
+    Fog::Compute.expects(:providers).at_least_once.returns([:aws, :libvirt])
```

Comment:
Is the count unstable?

Reply:
Yop, with `twice` it fails on
```
Failure:
ComputeResourceTest#test_0016_.providers returns merge of loaded builtin and registered providers [/home/lstejska/projects/foreman2/app/models/compute_resources/foreman/model/vmware.rb:27]
Minitest::Assertion: unexpected invocation: Fog::Compute.providers()
unsatisfied expectations:
- expected exactly twice, invoked 3 times: Fog::Compute.providers(any_parameters)
satisfied expectations:
- allowed any number of times, invoked never: #<AnyInstance:Resolv::DNS>.getaddresses(any_parameters)
- allowed any number of times, invoked never: #<AnyInstance:Resolv::DNS>.getaddress(any_parameters)
- allowed any number of times, invoked never: #<AnyInstance:Resolv::DNS>.getnames(any_parameters)
- allowed any number of times, invoked never: #<AnyInstance:Resolv::DNS>.getname(any_parameters)
- expected exactly once, invoked once: ComputeResource(id: integer, name: string, description: text, url: string, user: string, password: text, uuid: string, type: string, created_at: datetime, updated_at: datetime, attrs: text, caching_enabled: boolean, domain: string, http_proxy_id: integer).registered_providers(any_parameters)
```

Reply:
And with `.times(3)` instead of `.twice`?

Reply:
Fails with:
```
Failure:
ComputeResourceTest#test_0016_.providers returns merge of loaded builtin and registered providers [/home/lstejska/projects/foreman2/app/models/compute_resources/foreman/model/openstack.rb:23]
Minitest::Assertion: unexpected invocation: Fog::Compute.providers()
unsatisfied expectations:
- expected exactly 3 times, invoked 4 times: Fog::Compute.providers(any_parameters)
satisfied expectations:
- allowed any number of times, invoked never: #<AnyInstance:Resolv::DNS>.getaddresses(any_parameters)
- allowed any number of times, invoked never: #<AnyInstance:Resolv::DNS>.getaddress(any_parameters)
- allowed any number of times, invoked never: #<AnyInstance:Resolv::DNS>.getnames(any_parameters)
- allowed any number of times, invoked never: #<AnyInstance:Resolv::DNS>.getname(any_parameters)
- expected exactly once, invoked once: ComputeResource(id: integer, name: string, description: text, url: string, user: string, password: text, uuid: string, type: string, created_at: datetime, updated_at: datetime, attrs: text, caching_enabled: boolean, domain: string, http_proxy_id: integer).registered_providers(any_parameters)


45 tests, 81 assertions, 1 failures, 0 errors, 0 skips

```